File: examples/mcp/tools-only-ts/server.test.mjs
import assert from "node:assert/strict";

import {
  createMcpTestHarness,
  sequenceIdGenerator,
} from "../../../ts/dist/index.js";

import { createToolsOnlyMcpServer } from "./server.mjs";

const server = createToolsOnlyMcpServer({
  idGenerator: sequenceIdGenerator(["sess-example"], "sess"),
});
const harness = createMcpTestHarness(server);

const init = await harness.initialize({ id: "init" });
assert.equal(init.response.status, 200);
assert.equal(init.response.headers["mcp-session-id"]?.[0], "sess-example");
assert.deepEqual(init.bodyJson.result.capabilities, { tools: {} });

const sessionId = init.response.headers["mcp-session-id"][0];
const call = await harness.call(
  sessionId,
  "tools/call",
  { name: "echo", arguments: { message: "hello TypeScript MCP" } },
  "call",
);
assert.equal(call.response.status, 200);
assert.deepEqual(call.bodyJson.result, {
  content: [{ type: "text", text: "hello TypeScript MCP" }],
});

File: examples/mcp/tools-only-py/server_test.py
from __future__ import annotations

import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[3]
sys.path.insert(0, str(ROOT / "py" / "src"))
sys.path.insert(0, str(Path(__file__).resolve().parent))

from apptheory import create_mcp_test_harness, sequence_mcp_id_generator  # noqa: E402
from server import create_tools_only_mcp_server  # noqa: E402


def main() -> None:
    server = create_tools_only_mcp_server(
        {"id_generator": sequence_mcp_id_generator(["sess-example"], "sess")}
    )
    harness = create_mcp_test_harness(server)

    init = harness.initialize(id="init")
    assert init.response.status == 200
    assert init.response.headers["mcp-session-id"] == ["sess-example"]
    assert init.body_json["result"]["capabilities"] == {"tools": {}}

    session_id = init.response.headers["mcp-session-id"][0]
    call = harness.call(
        session_id,
        "tools/call",
        {"name": "echo", "arguments": {"message": "hello Python MCP"}},
        "call",
    )
    assert call.response.status == 200
    assert call.body_json["result"] == {
        "content": [{"type": "text", "text": "hello Python MCP"}]
    }

    print("examples/mcp/tools-only-py/server_test.py: PASS")


if __name__ == "__main__":
    main()
